packages/astro/src/core/ssr/html.ts

```diff
@@ -4,53 +4,68 @@ import htmlparser2 from 'htmlparser2';
 
 /** Inject tags into HTML (note: for best performance, group as many tags as possible into as few calls as you can) */
 export function injectTags(html: string, tags: vite.HtmlTagDescriptor[]): string {
-  // TODO: this usually takes 5ms or less, but if it becomes a bottleneck we can create a WeakMap cache
   let output = html;
   if (!tags.length) return output;
 
   const pos = { 'head-prepend': -1, head: -1, 'body-prepend': -1, body: -1 };
 
-  try {
```

Comment:
Got rid of the `try/catch` here because we actually should throw on invalid HTML; we should never receive it.